### PR TITLE
Add gnmi test to verify subscribe API

### PR DESCRIPTION
**Pull request by ganglyu**
### Description of PR

Summary:
Need new test to verify gnmi-native subscribe API.
Fixes # (issue)
Microsoft ADO: 25282671

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
Need new test to verify gnmi-native subscribe API.

#### How did you do it?
Add test to verify subscribe polling mode with DEVICE_METADATA table in CONFIG_DB.
Add test to verify subscribe streaming sample mode with DEVICE_METADATA table in CONFIG_DB.
Add test to verify subscribe streaming onchange mode with DEVICE_METADATA table in CONFIG_DB.
Add test to read COUNTERS_QUEUE_NAME_MAP table in COUNTERS_DB.
Add test to read COUNTERS table in COUNTERS_DB.
Add test to verify subscribe polling mode with COUNTERS_PORT_NAME_MAP table in COUNTERS_DB.
Add test to verify subscribe streaming sample mode with COUNTERS_PORT_NAME_MAP table in COUNTERS_DB.
Add test to verify subscribe polling mode with COUNTERS table in COUNTERS_DB.
Add test to verify subscribe streaming sample mode with COUNTERS table in COUNTERS_DB.

#### How did you verify/test it?
Run gnmi end to end test

### CODE REVIEW

File: tests/gnmi/helper.py [hunk 1]
```diff
@@ -234,6 +234,124 @@ def gnmi_get(duthost, ptfhost, path_list):
...
+        interval_ms: interval, unit is ms
+        count: update count
+
+    Returns:
+        msg: gnmi client output
+    """
+    env = GNMIEnvironment(duthost, GNMIEnvironment.GNMI_MODE)
+    ip = duthost.mgmt_ip
+    port = env.gnmi_port
+    interval = interval_ms / 1000
```
Comment (qiluo-msft) [inline tests/gnmi/helper.py:256]: >/ [](http://example.com/codeflow?start=27&length=1)

Is your intention integer division or float division?

File: tests/gnmi/helper.py [hunk 2]
```diff
@@ -234,6 +234,124 @@ def gnmi_get(duthost, ptfhost, path_list):
...
+    cmd = "docker exec %s gnmi_cli -client_types=gnmi -a %s:%s " % (env.gnmi_container, ip, port)
+    cmd += "-client_crt /etc/sonic/telemetry/gnmiclient.crt "
+    cmd += "-client_key /etc/sonic/telemetry/gnmiclient.key "
+    cmd += "-ca_crt /etc/sonic/telemetry/gnmiCA.pem "
+    cmd += "-logtostderr "
+    # Use sonic-db as default origin
+    cmd += '-origin=sonic-db '
+    cmd += '-query_type=polling '
+    cmd += '-polling_interval %us -count %u ' % (interval, count)
+    for path in path_list:
```
Comment (qiluo-msft) [inline tests/gnmi/helper.py:267]: >path_list [](http://example.com/codeflow?start=16&length=9)

check None?

File: tests/gnmi/helper.py [hunk 3]
```diff
@@ -234,6 +234,124 @@ def gnmi_get(duthost, ptfhost, path_list):
...
+        path = path.replace('sonic-db:', '')
+        cmd += '-q %s ' % (path)
+    output = duthost.shell(cmd, module_ignore_errors=True)
+    return output['stdout'] + " " + output['stderr']
```
Comment (qiluo-msft) [inline tests/gnmi/helper.py:271]: >return [](http://example.com/codeflow?start=4&length=6)

better to return a tuple?

File: tests/gnmi/helper.py [hunk 4]
```diff
@@ -234,6 +234,124 @@ def gnmi_get(duthost, ptfhost, path_list):
...
+    for path in path_list:
+        path = path.replace('sonic-db:', '')
+        cmd += " " + path
+    output = ptfhost.shell(cmd, module_ignore_errors=True)
+    msg = output['stdout'].replace('\\', '')
+    mark = 'response received:'
+    if mark in msg:
+        return msg
+    else:
+        return output['stderr'] + " " + msg
```
Comment (qiluo-msft) [inline tests/gnmi/helper.py:312]: >return [](http://example.com/codeflow?start=8&length=6)

better to return a tuple?

File: tests/gnmi/helper.py [hunk 5]
```diff
@@ -234,6 +234,124 @@ def gnmi_get(duthost, ptfhost, path_list):
...
```
Comment (qiluo-msft) [inline tests/gnmi/helper.py:352]: >return [](http://example.com/codeflow?start=8&length=6)

better to return a tuple?

File: tests/gnmi/helper.py (hunk 1 above)
Comment (ganglyu) [reply tests/gnmi/helper.py]: Fixed.

File: tests/gnmi/helper.py (hunk 2 above)
Comment (ganglyu) [reply tests/gnmi/helper.py:270]: Updated.

File: tests/gnmi/helper.py (hunk 3 above)
Comment (ganglyu) [reply tests/gnmi/helper.py]: Updated.

File: tests/gnmi/helper.py (hunk 4 above)
Comment (ganglyu) [reply tests/gnmi/helper.py]: Updated

File: tests/gnmi/helper.py (hunk 5 above)
Comment (ganglyu) [reply tests/gnmi/helper.py]: Updated.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**